### DISCUSSION TITLE
It's possible a only-fflate zip-no-worker-inflate.min.js version?

### DISCUSSION THREAD

**Discussion by juandavm** [gildas-lormeau/zip.js#248] [Q&A]
It's possible a dist file with only-fflate-based zip-no-worker-inflate.min.js version? Without the default and, in that case unused, inflate code.

The goal is to reduce js file size in a web environment and the zip processing time simultaneously.
Thanks in advance.

**Comment by 101arrowz** [comment]
If you're aiming to minimize bundle size, I'd like to mention you can use `fflate` directly, as it provides ZIP APIs already. `zip.js` has some nice abstractions, support for crypto, support for Zip64 compression, etc. but if you just need basic ZIP support and Zip64 decompression, `fflate` alone will yield smaller bundle size. Check near the bottom of the [usage section of the README](https://github.com/101arrowz/fflate/blob/master/README.md#usage) if you'd like to try it.

Also, this yields support for tree-shaking, meaning you will be able to omit the unused code automatically if you use a bundler.

**Comment by gildas-lormeau** [comment]
You have to download the project and run:
```
npm install
npm install fflate
npx rollup -c rollup-fflate.config.js
```

The `dist/zip-no-worker-inflate.min.js` file will use fflate.

**Comment by juandavm** [comment]
I have done what you said @gildas-lormeau and the compilation works correctly but at the execution time it throws an error:

```
Uncaught (in promise) TypeError: e.inflate.append is not a function
    at qt.append (worker.js:1)
    at Object.append (worker.js:1)
    at i (worker.js:1)
    at async fe.getData (worker.js:1)
append @ worker.js:1
append @ worker.js:1
i @ worker.js:1
Promise.then (async)
(anonymous) @ worker.js:95
cacheEntry @ worker.js:94
(anonymous) @ worker.js:75
Promise.then (async)
(anonymous) @ worker.js:74
cacheContents @ worker.js:71
(anonymous) @ worker.js:198
```

The error is in this call:
`return await r.append(e)`

On this function:

```
async append(t) {
            const e = this;
            return e.encrypted && t.length && (t = await e.decrypt.append(t)),
            e.compressed && t.length && (t = await e.inflate.append(t)),   <<------- here  e.inflate.append is not a function
            (!e.encrypted || e.zipCrypto) && e.signed && t.length && e.crc32.append(t),
            t
        }
```

With the standalone version it works well, any incompatibility with fflate?

**Comment by gildas-lormeau** [comment]
Thank you @juandavm for the feedback. The issue should be fixed now.
